_posts/2019/2019-10-20-use-a-deprecation-message.md

```diff
@@ -0,0 +1,114 @@
+---
+title: "Use Active Support in Rails for deprecation messages"
+description: "An internal Rails tool, availible for us"
+layout: article
+category: ruby
+image:
+  base: '2019/use-active-support-in-rails-for-deprecation-messages'
+  alt: 'Caution on the pavement'
+  credit: Goh Rhy Yan
+  source: "https://unsplash.com/photos/FFgcWvplwsc"
+
+---
+
+When authors of open source libraries change major functionality they often adhere to something called [semantic versioning](https://semver.org). At its simplest you can think of it as a rule meaning features aren’t removed until a major version is released.
```

Comment:
But not necessarily...it can be the opposite: a new major version can just mean a load of new features.

So maybe say: When it comes to removing features, this isn't done until a major... or something like that?

But maybe you think the wording is fine!

Reply:
Comma after 'major functionality' in first line.

Reply:
> But not necessarily...it can be the opposite: a new major version can just mean a load of new features.

Technically if it's only new, non-breaking functionality it can be a point release. Rails tend just to bundle up new features with refactors and deprecations.

Reply:
Okay then!